internal/core/src/clustering/Clustering.h

```diff
@@ -0,0 +1,47 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+#include <vector>
+#include <unordered_map>
+#include <string>
+#include "common/Types.h"
+
+namespace milvus::clustering {
+
+// after clustering result uploaded, return result meta for golang usage
+struct ClusteringResultMeta {
+    std::string centroid_path;   // centroid result path
+    int64_t centroid_file_size;  // centroid result size
+    std::unordered_map<std::string, int64_t>
+        id_mappings;  // id mapping result path/size for each segment
+};
+
+class Clustering {
```

Comment:
We don't need polymorphism here. Since Knowhere has already supported a polymorphism for different clustering algo (Kmeans for instance). 
If we need one more layer here, it must be some other purpose like Mem-clustering, Disk-clustering. 
For now, I suggest we can simply don't use it.

Reply:
But when we use cgo, we will cast the class pointer to void* for golang usage and cast back in c code, we will lose the class template information, I think we could use the virtual class here and make the kmeansClustering -> MemClustering change